tests/integration/release/test_release.py

```diff
@@ -317,3 +331,60 @@ def test_guardian_duplication(
                         ]
 
             assert dependent[column] in guardians_values
+
+
+def test_dataset_missingness(
+    dataset_params: tuple[
+        str,
+        Callable[..., pd.DataFrame],
+        str | None,
+        int | None,
+        str | None,
+        Literal["pandas", "dask"],
+    ],
+    dataset_name: str,
+    mocker: MockerFixture,
+) -> None:
+    """Tests that missingness is accurate with dataset.data."""
+    mocker.patch(
+        "pseudopeople.dataset.Dataset.drop_non_schema_columns", side_effect=lambda df, _: df
+    )
+
+    # create unnoised dataset
+    _, dataset_func, source, year, state, engine = dataset_params
+    kwargs = {
+        "source": source,
+        "config": NO_NOISE,
+        "year": year,
+        "engine": engine,
+    }
+    if dataset_func != generate_social_security:
+        kwargs["state"] = state
+    unnoised_data = dataset_func(**kwargs)
+
+    # We must manually clean the data for noising since we are recreating our main noising loop
+    dataset_schema = DATASET_SCHEMAS.get_dataset_schema(dataset_name)
+    dataset = Dataset(dataset_schema, unnoised_data, SEED)
+    dataset._clean_input_data()
+    # convert datetime columns to datetime types for _reformat_dates_for_noising
+    for col in [COLUMNS.dob.name, COLUMNS.ssa_event_date.name]:
+        if col in dataset.data:
+            dataset.data[col] = pd.to_datetime(dataset.data[col])
+            dataset.data["copy_" + col] = pd.to_datetime(dataset.data["copy_" + col])
+    dataset._reformat_dates_for_noising()
+    config = get_configuration()
+
+    # NOTE: This is recreating Dataset._noise_dataset but adding assertions for missingness
+    for noise_type in NOISE_TYPES:
+        if isinstance(noise_type, RowNoiseType):
+            if config.has_noise_type(dataset.dataset_schema.name, noise_type.name):
+                noise_type(dataset, config)
+                # Check missingness is synced with data
```

Comment:
"Check that dataset.missingness was updated correctly by noising function to match noised data"